Stack/Opc.Ua.Core/Security/Certificates/CertificateFactory.cs

```diff
@@ -415,9 +444,326 @@ string password
 
         return certificate;
     }
-    #endregion
 
-    #region Private Methods
+    /// <summary>
+    /// Revoke the CA signed certificate. 
+    /// The issuer CA public key, the private key and the crl reside in the storepath.
+    /// The CRL number is increased by one and existing CRL for the issuer are deleted from the store.
+    /// </summary>
+    public static async Task RevokeCertificateAsync(
```

Comment:
Shouldn't this method be in the X509CRL file?

Reply:
'Creates' a CRL, so its rather in factory